File: vortexdbg-android/src/main/java/com/vortexdbg/hook/xhook/IxHook.java
package com.vortexdbg.hook.xhook;

import com.vortexdbg.hook.IHook;
import com.vortexdbg.hook.ReplaceCallback;

/**
 * Only support android
 */
public interface IxHook extends IHook {

    int RET_SUCCESS = 0;

    void register(String pathname_regex_str, String symbol, ReplaceCallback callback);
    void register(String pathname_regex_str, String symbol, ReplaceCallback callback, boolean enablePostCall);

    void refresh();

}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/hook/xhook/IxHook.kt
package com.vortexdbg.hook.xhook

import com.vortexdbg.hook.IHook
import com.vortexdbg.hook.ReplaceCallback

/**
 * Only support android
 */
interface IxHook : IHook {

    fun register(pathname_regex_str: String, symbol: String, callback: ReplaceCallback)
    fun register(pathname_regex_str: String, symbol: String, callback: ReplaceCallback, enablePostCall: Boolean)

    fun refresh()

    companion object {
        const val RET_SUCCESS = 0
    }

}
